lldb/test/API/functionalities/data-formatter/type-recognizer/value-object-cast/RecognizerFormatter.py

```diff
@@ -0,0 +1,74 @@
+from lldb import SBType, SBValue
+
+def generalHandler(value: SBValue, internal_dict) -> SBType:
+    supported_targets = [
+        "A_1",
+        "A_2",
+
+        "B_2",
+        "B_3",
+    ]
+
+    # Get target type from value data members
+    assert value.GetChildMemberWithName("target").IsValid()
+    target_type: SBValue = value.GetChildMemberWithName("target").value
+
+    if target_type in supported_targets:
+        derived_type: SBType = value.target.FindFirstType(target_type)
+    else:
+        raise NotImplementedError(f"Non-supported target type: {target_type}")
+
+    assert derived_type.IsValid()
+    if value.type.IsPointerType():
+        derived_type = derived_type.GetPointerType()
+
+    return derived_type
+
+def handleVirtualDerivatives(value: SBValue, internal_dict) -> SBType:
+    supported_targets = [
+        "A_1",
+        "A_2",
+    ]
+
+    # Get intermediate base from value children
```

Comment:
It would be nice to have this code closer to the actual test, and to inheritance graph associated with it. I remember that commands that take names of Python function also let you type in Python code. Can we leverage that?